cdap-data-fabric/src/main/java/io/cdap/cdap/spi/data/common/DefaultStorageProvider.java

```diff
@@ -45,11 +48,17 @@ public class DefaultStorageProvider implements StorageProvider {
   @Inject
   DefaultStorageProvider(Injector injector, CConfiguration cConf, SConfiguration sConf,
                          StorageProviderExtensionLoader extensionLoader,
-                         MetricsCollectionService metricsCollectionService) {
+                         MetricsCollectionService metricsCollectionService,
+                         @Named(StorageModule.STORAGE_IMPL_NAME_KEY) String storageNameKey) {
```

Comment:
I'm not following what's happening here. Why does the key for the default need to be injected like this instead of forcing the cconf to have the right value in there?

Reply:
This change is to extend the DefaultStorageProvider class to support subcomponent-specific storage mechanisms (i.e. preview runners should never use the system metadata store even if the global metadata store is defined) and follows a similar pattern as the [RemoteAuthenticatorProvider](https://github.com/cdapio/cdap/blob/046dae65a5292c47711db2b8be81ec09839a4b70/cdap-common/src/main/java/io/cdap/cdap/common/internal/remote/DefaultRemoteAuthenticatorProvider.java). However, it's not used in this PR because we removed a bunch of modules that required the storage module in the first place, and that functionality is instead implemented in the [PreviewConfigModule](https://github.com/cdapio/cdap/blob/e5141e954f77b44c102dcc1f700142017f3f8e59/cdap-app-fabric/src/main/java/io/cdap/cdap/app/preview/PreviewConfigModule.java) in which several cConf settings are overwritten.

Reply:
Since this functionality is not being used, I've gone ahead and removed it.